### PR TITLE
Unit test crash in edm::checkForModuleDependencyCorrectness

### PR DESCRIPTION
**Pull request by fwyzard**
#### PR description:

Support arbitrary path names in the TriggerResultsFilter in HLTPathStatus mode

Other changes:
  - clean up and modernise the module
  - add a new unit test for the HLTPathStatus mode

#### PR validation:

The new unit tests should run.

### CODE REVIEW

**Comment by fwyzard** [comment]
please test

**Comment by fwyzard** [comment]
Please note that this PR is known to be broken; it should be kept as a test case for #36726 .

File: HLTrigger/HLTcore/interface/TriggerExpressionData.h [hunk 1]
```diff
@@ -87,15 +87,15 @@ namespace triggerExpression {
          edm::InputTag const& l1tResultsTag,
          bool l1tIgnoreMaskAndPrescale,
          bool doThrow,
-         edm::ConsumesCollector&& iC)
+         edm::ConsumesCollector&& cc)
```
Comment (Dr15Jones) [inline HLTrigger/HLTcore/interface/TriggerExpressionData.h:90]: FYI we changed this so you can pass a copy of a ConsumesCollector which avoids some weirdnesses with using `&&`.

**Comment by Dr15Jones** [comment]
@fwyzard could you change the pull request name so as not to confuse it with #36732 ? Thanks.

**Comment by fwyzard** [comment]
Good point @Dr15Jones , changed to make it explicit that this can be used to validate the fix for the crash.

**Comment by fwyzard** [comment]
please test

**Comment by Dr15Jones** [comment]
Please test

**Comment by Dr15Jones** [comment]
The unit test failure is no longer because of a segmentation fault but now is from the proper exception reporting on the unrunnable schedule.

**Comment by fwyzard** [comment]
```
----- Begin Fatal Exception 22-Jan-2022 16:12:25 CET-----------------------
An exception of category 'ScheduleExecutionFailure' occurred while
   [0] Calling beginJob
Exception Message:
Unrunnable schedule
The Path/EndPath configuration could cause the job to deadlock
  module 'filter_any_doublestar' is on path 'Check_Any_Doublestar' and depends on module 'Check_Any_Doublestar'
  module 'Check_Any_Doublestar' is on path 'Check_Any_Doublestar' and follows module 'filter_any_doublestar' on the path
----- End Fatal Exception -------------------------------------------------
```

**Comment by fwyzard** [comment]
Thanks @Dr15Jones .

**Pull request closed without merging**